WcaOnRails/app/views/results/_results_selector.html.erb

```diff
@@ -0,0 +1,153 @@
+        <% end %>
+      </ul>
+      <label id="until-years" class="btn btn-info <%= until_active ? " active" : "" %>" data-toggle="dropdown">
+        <span class="caption">
+          <% if until_active %>
+            <%= t("results.selector_elements.years_selector.until_html", year: @year) %>
+          <% else %>
+            <%= t("results.selector_elements.years_selector.until") %>
+          <% end %>
+        </span>
+      </label>
+    </div>
+  </div>
+</div>
+
+<% if show_rankings_options %>
+  <div class="form-group">
+    <%= label_tag(t("results.selector_elements.show_selector.show")) %>
+    <div id="show" class="btn-group">
+      <div class="btn-group">
+        <% persons_active = @is_persons %>
+        <label id="persons" class="btn btn-info <%= persons_active ? " active" : "" %>" data-toggle="dropdown">
+          <span class="caption">
+            <% if persons_active %>
+              <%= t("results.selector_elements.show_selector.persons_html", quantity: @show) %>
+            <% else %>
+              <%= t("results.selector_elements.show_selector.persons") %>
+            <% end %>
+          </span>
+        </label>
+        <ul class="dropdown-menu show-persons">
+          <% @quantities.each do |quantity| %>
+            <li class="persons <%= (persons_active && quantity.to_s == @show ? " active" : "") %>"><%= link_to quantity.to_s, request.params.merge(show: "#{quantity} persons") %></li>
+          <% end %>
+        </ul>
+      </div>
+      <div class="btn-group">
+        <% results_active = @is_results %>
+        <label id="results" class="btn btn-info <%= results_active ? " active" : "" %>" data-toggle="dropdown">
+          <span class="caption">
+            <% if results_active %>
+              <%= t("results.selector_elements.show_selector.results_html", quantity: @show) %>
+            <% else %>
+              <%= t("results.selector_elements.show_selector.results") %>
+            <% end %>
+          </span>
+        </label>
+        <ul class="dropdown-menu show-results">
+          <% @quantities.each do |quantity| %>
+            <li class="results <%= (results_active && quantity.to_s == @show ? " active" : "") %>"><%= link_to quantity.to_s, request.params.merge(show: "#{quantity} results") %></li>
+          <% end %>
+        </ul>
+      </div>
+      <%= link_to t("results.selector_elements.show_selector.by_region"), request.params.merge(show: "by region"), class: "btn btn-info" + (@is_by_region ? " active" : "") %>
+    </div>
+  </div>
+<% elsif show_records_options %>
+  <div class="form-group">
+    <%= label_tag(t("results.selector_elements.show_selector.show")) %>
+    <div id="show" class="btn-group">
+      <% @shows.each do |show| %>
+        <%= link_to t("results.selector_elements.show_selector.#{show.gsub(" ", "_")}"),
```

Comment:
For simple char to char replacement please use [tr](https://apidock.com/ruby/String/tr) instead of gsub, since it avoids going through regular expression stuff.
On my computer gsub is roughly 3.5x slower than tr ;)

Reply:
Interesting, changed!